test/integration/deep/opaqueports/assertions_test.go

```diff
@@ -0,0 +1,75 @@
+package opaqueports
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var (
+	tcpMetricRE = regexp.MustCompile(
```

Comment:
A parsing library for those metrics would be cool. It'd be in favour of having an util in TestHelper that can help us make sense of the proxy metrics; definitely shouldn't be scoped to this PR though.

I'd be in favour of rolling those regex assertions into one of the test files. Very subjective, but think at the very least we should remove `_test` from its name so the test runner doesn't try to compile/run it, or whatever it is that Go's runner does in those situations.

Reply:
I submitted this issue: https://github.com/linkerd/linkerd2/issues/8250

Reply:
As for:
> I'd be in favour of rolling those regex assertions into one of the test files. Very subjective, but think at the very least we should remove _test from its name so the test runner doesn't try to compile/run it, or whatever it is that Go's runner does in those situations.

I think Go compiler will always compile it when running tests, regardless of `_test` suffix. What is important is that these regex are required for running tests.

Reply:
There are no tests in here, so I think it makes more to call this something like `util.go` for now since it will already be in the `opaqueports` integration test package. Even if just for the developer, it's easier to realize there are no actual tests in here; I would expect a `..._test.go` file to have tests in it.

That being said, it's only used by `opaque_ports_test.go` so I'm also fine just moving things back to a single file. I don't think introducing a separate utility file that is still only used by one other file to be significantly clearer.

Reply:
@kleimkuhler How about calling this file `assertions.go`? Or `checks.go`? 

Since this file contains checks/assertions for metrics, I would like to avoid a generic name like "util".

Reply:
Friendly ping @kleimkuhler

Reply:
Rename done in https://github.com/linkerd/linkerd2/pull/8225/commits/8d9fc2f0a181826356a144bc48851bcc6d568759